src/annotation.js

```diff
@@ -150,25 +150,31 @@ function draw(chart, caller, clip) {
     area = chartArea;
   }
 
-  drawElements(chart, visibleElements, caller, area);
+  const drawableElements = drawElements(visibleElements, caller, area).sort((a, b) => a.element.options.z - b.element.options.z);
+
+  for (const item of drawableElements) {
+    item.element.draw(chart.ctx, item.area);
+  }
 
   if (clip) {
     unclipArea(ctx);
   }
 }
 
-function drawElements(chart, elements, caller, area) {
+function drawElements(elements, caller, area) {
```

Comment:
Would rename the function to something like `getDrawableElements` since in its current implementation it does not do the actual drawing

Reply:
yeah, make a lot of sense

Reply:
changed in getDrawableElements